awscrt/common.py

```diff
@@ -18,3 +18,23 @@ def get_cpu_count_for_group(group_idx: int) -> int:
     Returns number of processors in a given group.
     """
     return _awscrt.get_cpu_count_for_group(group_idx)
+
+class SystemEnvironment:
+
+    def __init__(self):
+        self._env = _awscrt.load_system_environment()
+
+        if self.is_ec2_nitro_instance():
+            self._detected_instance_type = _awscrt.get_ec2_instance_type(self._env)
+        else:
+            self._detected_instance_type = None       
+
+    def is_ec2_nitro_instance(self) -> bool:    
+        return _awscrt.is_env_ec2(self._env)
+    
+    def get_ec2_instance_type(self) -> str:        
+        return self._detected_instance_type
+    
+    def is_crt_s3_optimized_for_system_env(self) -> bool:
```

Comment:
This is awkward:
`system_env.is_crt_s3_optimized_for_system_env()`

flip the words around to sound more natural:
`system_env.is_optimized_for_crt_s3()`

or it might feel more natural as a separate function in s3.py:
`awscrt.s3.is_optimized_for_system(system_env)`

Reply:
Or just have 1 public function: `awscrt.s3.is_optimized_for_system()`

and don't bother binding the aws_system_environment at all, just keep it as an implementation detail

we can have python cache the answer, so it only does work the first time it's called

Reply:
That's fine. I added verbosity here largely for aid in debugging when something goes wrong, we can at least ask the customer to add a python function to their script and it's much simpler to track down, than having to peel through logs or c source.